gprofiler/python.py

```diff
@@ -159,6 +159,20 @@ def _check_output(cls, process, output_path):
             cls._check_missing_headers(stdout)
             raise CalledProcessError(process.returncode, process.args, stdout, stderr)
 
+    @classmethod
+    def get_pyperf_cmd(cls) -> List[str]:
+        pyperf = resource_path(cls.PYPERF_RESOURCE)
+        staticx_dir = os.getenv("STATICX_BUNDLE_DIR")
+        # are we running under staticx?
+        if staticx_dir is not None:
+            # STATICX_BUNDLE_DIR is where staticx has extracted all of the libraries it had collected
+            # earlier.
+            # see https://github.com/JonathonReinhart/staticx#run-time-information
+            # we run PyPerf with the same ld.so whose libc it was compiled against.
+            return [f"{staticx_dir}/.staticx.interp", "--library-path", staticx_dir, pyperf]
```

Comment:
why not `patchelf` when running staticx?

Reply:
No reason. I'll still need to pass `LD_LIBRARY_PATH` tho so it doesn't allow getting rid of the code here (unless we patch that as well?)

Reply:
Nope. Use `--set-rpath '$ORIGIN/../..../' --force-rpath`. Passing `LD_LIBRARY_PATH` would mess up any programs spawned by pyperf (because they would inherit it).

Reply:
Well, any program spawned by PyPerf should get its `LD_LIBRARY_PATH` as well, to use any libs that bcc might want (and in general, that's the only directory any program should be loaded libraries from).

I'll try the `patchelf`.

Reply:
We can modify staticx to do that (it already does it for the main executable)

Reply:
Talked IRL - `$ORIGIN` won't do due to:

> Since the path is determined at runtime, that's not possible (unless we hardcode it by modifying staticx)

(and they're not in the same directory)